module/zfs/qat_crypt.c

```diff
@@ -557,10 +577,56 @@ qat_checksum(uint64_t cksum, uint8_t *buf, uint64_t size, zio_cksum_t *zcp)
 	return (status);
 }
 
-module_param(zfs_qat_encrypt_disable, int, 0644);
-MODULE_PARM_DESC(zfs_qat_encrypt_disable, "Disable QAT encryption");
+static int
+param_set_qat_encrypt(const char *val, struct kernel_param *kp)
+{
+	int ret;
+	int *pvalue = kp->arg;
+	ret = param_set_int(val, kp);
+	if (ret)
+		return (ret);
+	/*
+	 * zfs_qat_encrypt_disable = 0: enable qat encrypt
+	 * try to initialize qat instance if it has not been done
+	 */
+	if (*pvalue == 0 && !qat_cy_init_done) {
+		ret = qat_cy_init();
+		if (ret != 0) {
+			zfs_qat_encrypt_disable = 1;
+			return (ret);
+		}
+	}
```

Comment:
I'm having a hard time finding the source code for `param_set_int()`, but I assume you have tested that this functionality works?

Reply:
For reference, the `param_get|set_*` functions are provided by the kernel in `include/linux/moduleparam.h`.  We use  `param_set_int` in `module/zfs/spa_misc.c` for updating the `spa_slop_shift` in the same fashion.